### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Nikoramo**
- [DEMO LINK](https://Nikoramo.github.io/js_2048_game/)

### CODE REVIEW

File: src/scripts/main.js [hunk 1]
```diff
@@ -1,3 +1,238 @@
...
+
+function needAddNewCell(arrayRow, direction) {
+  return arrayRow.some(item => {
+    switch (direction) {
+      case 'ArrowLeft':
+        for (let i = item.children.length - 1; i > 0; i--) {
+          if (item.children[i].textContent
+            && !item.children[i - 1].textContent) {
+            return true;
+          }
```
Comment (sTorba24) [inline src/scripts/main.js:208]: so return just condition. If it's true you return true
```suggestion
          return item.children[i].textContent && !item.children[i - 1].textContent;
```

File: src/scripts/main.js [hunk 2]
```diff
@@ -1,3 +1,238 @@
...
+    case 'ArrowRight':
+      for (let i = 0; i < sacrifice.length; i++) {
+        fieldRows[indexRow].append(sacrifice[i]);
+      }
+      break;
+  }
+
+  function sortCells(array) {
+    array.sort((a, b) => {
+      if (a.textContent === '') {
```
Comment (sTorba24) [inline src/scripts/main.js:177]:
```suggestion
      if (!a.textContent) {
```

Comment (sTorba24) [reply src/scripts/main.js:177]: fix everywhere

**Review by sTorba24 — APPROVED** [review]
Good work!

**Review by DarkMistyRoom — CHANGES_REQUESTED** [review]
I've found one moment about horizontal merging. Look at the bottom line when I click left only 2s should merge not all of the cells [VIDEO](https://www.loom.com/share/11f09a6ab95e4c3dbd8ccf3f58b7dc57?sid=cbbd1aac-cbe6-4d48-8a9b-fad22a3f9afb)

**Pull request closed without merging**